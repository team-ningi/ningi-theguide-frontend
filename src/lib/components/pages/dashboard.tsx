/* eslint-disable react-hooks/exhaustive-deps */
import { Box, Paragraph, Flex } from "theme-ui";
import { useEffect, useState } from "react";
import { connect } from "react-redux";
import { Dispatch } from "redux";
import { useRouter } from "next/navigation";
import {
  SessionType,
  SetLoadingType,
  UserType,
  DocType,
  DashboardStateType,
} from "@/lib/types";
import {
  FilePdf,
  FileText,
  CheckCircle,
  WarningCircle,
  CaretRight,
  CaretDown,
} from "phosphor-react";
import { getUserDocuments } from "@/utils/api-helper";
import { AddNewForm } from "@/lib/components/addContent/add-new-document";
import moment from "moment";

const IconMap = {
  txt: <FileText size={22} />,
  pdf: <FilePdf size={22} />,
};

export const TableHeader = () => (
  <Flex
    sx={{
      justifyContent: "space-between",
      alignItems: "center",
      borderBottom: "1px solid #E2E8F0",
      height: "50px",
      fontSize: "15px",
      p: "0 20px 10px",
      backgroundColor: "#F8F8F8",
      color: "#444",
      borderTopLeftRadius: "12px",
      borderTopRightRadius: "12px",
    }}
  >
    <Box
      sx={{
        textAlign: "left",
        pl: "27px",
        width: "100%",
        paddingTop: "15px",
        borderRight: "1px solid #E2E8F0",
        fontWeight: "600",
      }}
    >
      Label
    </Box>
    <Box
      style={{
        textAlign: "center",
        width: "130px",
        paddingTop: "15px",
        borderRight: "1px solid #E2E8F0",
        fontWeight: "600",
      }}
    >
      Type
    </Box>
    <Box
      style={{
        textAlign: "center",
        width: "130px",
        paddingTop: "15px",
        borderRight: "0px solid #E2E8F0",
        fontWeight: "600",
      }}
    >
      Status
    </Box>
  </Flex>
);

const FileKeyValue = ({
  theKey,
  theValue,
  isLink,
}: {
  theKey: string;
  theValue?: string;
  isLink?: string;
}) => (
  <Flex sx={{ width: "100%", ml: "26px", mb: "5px" }}>
    <Paragraph
      sx={{
        fontWeight: "300",
        fontSize: "14px",
        minWidth: "130px",
        textAlign: "right",
      }}
    >
      {theKey} :
    </Paragraph>

    {!isLink && (
      <Paragraph sx={{ fontWeight: "300", fontSize: "14px", ml: "20px" }}>
        {theValue}
      </Paragraph>
    )}
    {isLink && (
      <Paragraph
        sx={{
          fontWeight: "300",
          fontSize: "14px",
          ml: "20px",
          cursor: "pointer",
        }}
        onClick={() => window.location.assign(`${isLink}`)}
      >
        Download - put download icon here
      </Paragraph>
    )}
  </Flex>
);

const TableItem = ({ item, i }: { item: any; i: number }) => {
  const [showDetails, toggleDetails] = useState<boolean>(false);

  const { file_type, label, embedding_created } = item;
  const isEven = i % 2 === 0;

  return (
    <Flex
      sx={{
        p: "0 20px",
        borderBottom: "1px solid #E2E8F0",
        minHeight: showDetails ? "230px" : "70px",
        backgroundColor: !isEven ? "#F8F8F8" : "white",
        flexDirection: "column",
        justifyContent: "flex-start",
      }}
    >
      <Flex
        sx={{
          flex: 1,
          height: "100px",
          fontSize: "14px",
          justifyContent: "space-between",
          alignItems: "center",
        }}
      >
        <Flex
          sx={{
            textAlign: "left",
            ml: "30px",
            width: "100%",
            borderRight: "1px solid #E2E8F0",
            cursor: "pointer",
          }}
          onClick={() => toggleDetails(!showDetails)}
        >
          <Paragraph
            sx={{ fontWeight: "500", mr: "20px", textTransform: "capitalize" }}
          >
            {label}{" "}
          </Paragraph>
          {!showDetails && <CaretRight size={17} />}
          {showDetails && <CaretDown size={17} />}
        </Flex>
        <Box
          style={{
            textAlign: "center",
            width: "130px",
            borderRight: "1px solid #E2E8F0",
            fontWeight: "500",
            color: "#444",
          }}
        >
          {/* @ts-ignore */}
          {IconMap[file_type]}
        </Box>
        <Box
          style={{
            textAlign: "center",
            width: "130px",
            borderRight: "0px solid #E2E8F0",
            color: embedding_created ? "green" : "firebrick",
          }}
        >
          {embedding_created ? (
            <CheckCircle size={22} />
          ) : (
            <WarningCircle size={22} />
          )}
        </Box>
      </Flex>
      {showDetails && (
        <Flex sx={{ height: "160px", flexDirection: "column" }}>
          {/* moment */}
          <FileKeyValue
            theKey="Uploaded"
            theValue={moment(item.updated_at).format("dddd, MMMM Do YYYY")}
          />
          <FileKeyValue theKey="File Type" theValue={item.file_type} />
          <FileKeyValue theKey="Original File" isLink={item.file_url} />
          <FileKeyValue
            theKey="Original Filename"
            theValue={item.original_filename}
          />
          <FileKeyValue
            theKey="Vector Embedded"
            theValue={
              item.embedding_created
                ? "yes"
                : "failed (try again by clicking the button below)"
            }
          />
          {!item.embedding_created && (
            <Paragraph
              sx={{
                fontWeight: "500",
                width: "200px",
                mt: "10px",
                fontSize: "14px",
                cursor: "pointer",
              }}
              onClick={() => alert(item?._id)}
            >
              Embed document
            </Paragraph>
          )}
        </Flex>
      )}
    </Flex>
  );
};

const defaultState = {
  showUpload: false,
  showSuccess: false,
  user_id: "",
  customFilename: "",
  label: "",
};

const DashboardComponent = ({
  user,
  session,
  setLoading,
}: {
  user: UserType;
  session: SessionType;
  setLoading: SetLoadingType;
}) => {
  const [state, updateState] = useState<DashboardStateType>(defaultState);
  const [docs, updateDocs] = useState<DocType[]>([]);
  const router = useRouter();

  useEffect(() => {
    (async () => {
      setLoading(true);
      const { data } = await getUserDocuments(
        user._id,
        session?.authToken,
        "all"
      );

      updateDocs(data);
      updateState({ ...state, user_id: user?._id });
      console.log(data);
      setLoading(false);
    })();
  }, []);

  return (
    <Box
      className="sectionContainer"
      sx={{
        width: "900px",
        opacity: 1,
        mt: "20px",
        minHeight: "600px",
        textAlign: "center",
        backgroundColor: "transparent",
        border: "0px solid blue",
        ml: "0px",
        "@media screen and (max-width: 1275px)": { width: "600px" },
        "@media screen and (max-width: 700px)": {
          width: "100%",
          border: "0px solid red",
          ml: "auto",
          mr: "auto",
        },
        "@media screen and (max-width: 550px)": {
          width: "250px",
          ml: "60px",
        },
      }}
    >
      {docs?.length > 0 && (
        <Box
          style={{
            marginTop: "30px",
            border: "1px #E2E8F0 solid",
            width: "800px",
            minHeight: "50px",
            marginBottom: "20px",
            borderTopLeftRadius: "12px",
            borderTopRightRadius: "12px",
          }}
        >
          <TableHeader />
          {docs?.map((item: DocType, i: number) => (
            <TableItem item={item} i={i} key={`${item.label}`} />
          ))}
        </Box>
      )}

      {docs?.length < 1 && <Paragraph>Upload docs first</Paragraph>}

      <AddNewForm
        state={state}
        updateState={updateState}
        setLoading={setLoading}
        user={user}
        session={session}
      />
    </Box>
  );
};

const mapDispatchToProps = (dispatch: Dispatch) => ({
  setLoading: (loading: boolean) => dispatch({ type: "SET_LOADING", loading }),
});

const mapStateToProps = (state: { account: {} }) => {
  const { account } = state;
  return {};
};
export default connect(mapStateToProps, mapDispatchToProps)(DashboardComponent);
