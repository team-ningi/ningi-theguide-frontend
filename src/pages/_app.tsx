import "@/styles/globals.css";
import type { AppProps } from "next/app";
import { ThemeUIProvider } from "theme-ui";
import theme from "../lib/theme";
import { Poppins, Roboto } from "next/font/google";
import { Provider } from "react-redux"; //@ts-ignore
import { configureStore, compose } from "@reduxjs/toolkit";
import createSagaMiddleware from "redux-saga";
import reducers from "../store/reducers";
import { rootSaga } from "../store/sagas/index";

const poppins = Poppins({
  subsets: ["latin"],
  display: "swap",
  variable: "--font-poppins",
  weight: ["100", "200", "300", "400", "500", "600", "700", "800", "900"],
});
const roboto = Roboto({
  subsets: ["latin"],
  display: "swap",
  variable: "--font-roboto",
  weight: ["300", "400", "500", "700", "900"],
});

const sagaMiddleware = createSagaMiddleware();

const composer =
  typeof window !== "undefined" // @ts-ignore
    ? window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__ || compose
    : compose;

export const store = configureStore({
  reducer: reducers,
  middleware: [sagaMiddleware],
});
sagaMiddleware.run(rootSaga);

export default function App({ Component, pageProps }: AppProps) {
  return (
    <Provider store={store}>
      <ThemeUIProvider theme={theme}>
        <main className={roboto.className}>
          <Component {...pageProps} />
        </main>
      </ThemeUIProvider>
    </Provider>
  );
}
