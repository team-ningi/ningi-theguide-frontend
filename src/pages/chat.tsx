/* eslint-disable react-hooks/exhaustive-deps */

import Head from "next/head";
import { useRouter } from "next/navigation";
import { useEffect, useState } from "react";
import Wrapper from "@/lib/components/appWrapper";
import { Spinner } from "../lib/components/spinner";
import { ChatInterface } from "@/lib/components/chatInterface";
import serverSidePropsWithAuth from "../utils/server_side_props_with_auth";
import { PageTypes, SessionType, UserType } from "@/lib/types";
import { connect } from "react-redux";
import { Dispatch } from "redux";

const Chat = ({ session, setCoreData, user }: PageTypes) => {
  const router = useRouter();

  useEffect(() => {
    (async () => {
      if (session?.email) {
        if (!user) {
          await setCoreData(session.email, session, false);
        }
      } else {
        window.location.assign("/");
      }
    })();
  }, []);

  return (
    <>
      <Head>
        <title>Chat</title>
        <meta name="description" content="Ningi playground Dashboard" />
        <meta name="viewport" content="width=device-width, initial-scale=1" />
        <link rel="icon" href="/favicon.ico" />
      </Head>
      <Wrapper>
        <ChatInterface session={session} />
      </Wrapper>
    </>
  );
};

const mapDispatchToProps = (dispatch: Dispatch) => ({
  updateDarkMode: () => dispatch({ type: "TOGGLE_DARK_MODE" }),
  setCoreData: (
    email: string,
    session: SessionType,
    keepLoadingSpinner: boolean
  ) =>
    dispatch({
      type: "REFRESH_CORE_DATA",
      action: { email, session, keepLoadingSpinner },
    }),
});

const mapStateToProps = (state: {
  account: { darkMode: boolean; user: UserType };
}) => {
  const { account } = state;
  return {
    darkMode: account.darkMode,
    user: account.user,
  };
};
export default connect(mapStateToProps, mapDispatchToProps)(Chat);
export const getServerSideProps = serverSidePropsWithAuth({ redirect: true });
