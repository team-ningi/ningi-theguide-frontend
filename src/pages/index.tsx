/* eslint-disable react-hooks/exhaustive-deps */
import { useEffect, useState } from "react";
import { Dispatch } from "redux";
import { Flex, Box, Paragraph } from "theme-ui";
import Header from "@/lib/components/header";
import { Spinner } from "@/lib/components/spinner";
import { connect } from "react-redux";
import { LoginForm, Waiting, Verify } from "@/lib/components/login";
import { LoginTypes } from "@/lib/types";
import serverSidePropsWithAuth from "../utils/server_side_props_with_auth";

/*
// TODO
NEXT- >>> 
          - UPLOAD FILES TO S3  ✅
            - IF MP3 -> TRANSCRIBE AND EMBED ❌
            - ELSE   -> EMBED ✅
          - CREATE EMBEDDING PINECONE (WILL NEED TO LOOK AT WHAT LQ DID FOR BIG FILES) ✅
          - LIST OUT UPLOADS FOR THE USER ❌ - SELECT(s) FILE TO QUERY AGAINST ❌
          - WHEN CHAT MSG SENDS (replce in memeory VDB) ❌
            -> similarity search at pinecone > return array > send to open ai ✅


// WHEN LOG IN
//  -> IF NO NAME SET SHOW UI TO ADD NAME + UPLOAD AVATAR  ❌

// CHAT UI ✅

// SET UP PINECONE ✅

// UPLOAD FILES  ✅
// -> goes into database  ✅
//    -> Embed into pinecone ❌

//  TEXT       CHAT ✅
//  AUDIO      CHAT ✅
//  TEXTRACT   CHAT ❌
    https://medium.com/@hatemalimam/extract-text-and-data-from-any-document-using-amazon-textract-in-node-js-9a72136c6e64
    https://medium.com/@hatemalimam/extract-text-and-data-from-any-document-using-amazon-textract-in-node-js-9a72136c6e64

// REPORTING  ❌
// look at petes wireframes


// CYPRESS TESTING
// SET ENV VAR OF TEST_APPLICATION_TOKEN === 'utgig_E2E_uigikb98687_E2E_5645ry'
// if this is used bybass the manage checks ( set Authorization header to TEST_APPLICATION_TOKEN in test calls )
//    This will allow us to do an E2E
//        1 -> create test stuff
//        2 -> start the tests, click thru each page
//        3 -> teardown test stuff

*/

const defaultState = {
  username: "",
  mode: "login", //  login || verify || waiting
};

const Login = ({ setCoreData, darkMode, session }: LoginTypes) => {
  const [currentState, updateState] = useState(defaultState);
  const [loading, toggleLoading] = useState(false);

  useEffect(() => {
    (async () => {
      toggleLoading(true);
      try {
        const queryString = window.location.href;
        const decodedQueryString = decodeURIComponent(queryString);
        const url = new URL(decodedQueryString);
        const mode = url.searchParams.get("mode");
        const encoded = url?.search?.split("=")[2];

        if (encoded) {
          if (mode === "verify") {
            updateState({ ...currentState, mode: "verify" });
            await setCoreData(atob(encoded), session, false);

            if (atob(encoded) === "test@ningi.co.uk") {
              // TODO
              // TEST ROUTE
              //  window.location.assign("/tests");
            } else {
              window.location.assign("/discussions");
            }
          }
        }
      } catch (e) {
        //  swallow
      }
      toggleLoading(false);
    })();
  }, []);

  return (
    <Flex
      sx={{
        height: "100vh",
        backgroundColor: "#f9f9f9",
        flexDirection: "column",
        position: "relative",
      }}
    >
      <Header logoType="dark" />

      <Flex
        sx={{
          height: "100vh",
          backgroundColor: "#F9F9F9",
          flexDirection: "column",
          position: "relative",
        }}
      >
        <Box
          sx={{
            height: "100vh",
            width: "1050px",
            backgroundColor: darkMode ? "#23242A" : "#f9f9f9",
            flexDirection: "column",
            backgroundImage: `url(/bg.png)`,
            backgroundSize: "cover",
            position: "absolute",
            right: "0",

            "@media screen and (max-width: 1450px)": {
              width: "750px",
              backgroundPosition: "left",
            },
            "@media screen and (max-width: 1150px)": {
              width: "650px",
              backgroundPosition: "left",
            },
            "@media screen and (max-width: 950px)": {
              width: "600px",
            },
            "@media screen and (max-width: 550px)": {
              backgroundImage: "unset",
            },
          }}
        />

        {loading && <Spinner />}

        {currentState?.mode === "login" && (
          <LoginForm
            currentState={currentState}
            updateState={updateState}
            toggleLoading={toggleLoading}
          />
        )}

        {currentState?.mode === "verify" && (
          <Verify currentState={currentState} updateState={updateState} />
        )}

        {currentState?.mode === "waiting" && (
          <Waiting currentState={currentState} updateState={updateState} />
        )}
      </Flex>
    </Flex>
  );
};

const mapDispatchToProps = (dispatch: Dispatch) => ({
  setCoreData: (
    email: string,
    session: { email: string; authToken: string },
    keepLoadingSpinner: boolean
  ) =>
    dispatch({
      type: "REFRESH_CORE_DATA",
      action: { email, session, keepLoadingSpinner },
    }),
});

const mapStateToProps = (state: {
  account: { darkMode: boolean; loading: boolean };
}) => {
  const { account } = state;
  return {
    darkMode: account.darkMode,
  };
};
export default connect(mapStateToProps, mapDispatchToProps)(Login);
export const getServerSideProps = serverSidePropsWithAuth({ redirect: false });
