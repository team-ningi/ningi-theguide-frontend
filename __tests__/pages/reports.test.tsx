//@ts-nocheck
import React from "react";
import { Provider } from "react-redux";
import { fireEvent, render, screen, waitFor } from "@testing-library/react";
import "@testing-library/jest-dom";
import Reports from "../../src/pages/reports";
import { store } from "../../src/pages/_app";
import { act } from "react-dom/test-utils";
import configureStore from "redux-mock-store";

const initialState = {
  account: { darkMode: false, user: { email: "test@ningi.co.uk" } },
};
const mockStore = (state = initialState) => configureStore()(state);

jest.mock("../../src/utils/api-helper", () => {
  const theTags = {
    data: [],
  };
  const docs = {
    data: [],
  };
  const templates = {
    data: [],
  };
  const reports = {
    data: [],
  };
  return {
    getUserReports: jest.fn(() => Promise.resolve(reports)),
    getUserDocuments: jest.fn(() => Promise.resolve(docs)),
    getUserTemplates: jest.fn(() => Promise.resolve(templates)),
    getTags: jest.fn(() => Promise.resolve(theTags)),
  };
});

jest.mock("next/navigation", () => ({
  useRouter: () => {},
  usePathname: () => {},
  useParams: () => {},
  useSearchParams: () => ({
    get: (e) => e,
  }),
}));

jest.mock("next/font/google", () => ({
  Poppins: () => {},
  Roboto: () => {},
}));

describe("Reports", () => {
  it("Reports page is rendered", async () => {
    const { getByText, container, getAllByTestId } = render(
      <Provider store={mockStore(initialState)}>
        <Reports session={{ email: "test@ningi.co.uk", authToken: "" }} />
      </Provider>
    );

    expect(container).toMatchSnapshot();

    await waitFor(async () => {
      const Title = screen.getAllByText(/Generated Reports/i);
      expect(Title[0]).toBeInTheDocument();

      const SubTitle = screen.getAllByText(
        /Generate personalized reports by importing a base template and utilizing data from your uploaded documents./i
      );
      expect(SubTitle[0]).toBeInTheDocument();

      const createReport = screen.getAllByText(/Create Report/i);
      expect(createReport[0]).toBeInTheDocument();

      fireEvent.click(createReport[0]);

      const ViewBtn = screen.getAllByText(/View Reports/i);
      expect(await ViewBtn[0]).toBeInTheDocument();
    });
  });
});
