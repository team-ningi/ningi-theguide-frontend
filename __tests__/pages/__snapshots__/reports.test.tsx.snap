// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Reports Reports page is rendered 1`] = `
<div>
  <div
    class="css-bd26up-Flex"
  >
    <div
      class="css-sgv2ec-Flex"
    >
      <div
        class="css-1u9w373-Box"
      >
        <img
          alt=" logo"
          class="css-5gqyw6-Box"
          height="auto"
          src="/ningi-logo-light.webp"
          width="120px"
        />
      </div>
    </div>
    <div
      class="css-vd02hx-Flex"
    >
      <div
        class="css-f68wjp-Flex"
      >
        <div
          class="css-1rvitfq-Flex"
        >
          <div
            class="css-vii8pg-Flex"
          >
            <div
              class="css-14p1f3h-Flex"
            >
              <div
                class="css-4s1fvz-Flex"
              >
                <div
                  class="css-kvl0hh-Box"
                >
                  <svg
                    fill="currentColor"
                    height="24"
                    viewBox="0 0 256 256"
                    width="24"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <rect
                      fill="none"
                      height="256"
                      width="256"
                    />
                    <path
                      d="M213.4,109.6l-80-72.7a8,8,0,0,0-10.8,0l-80,72.7a8.3,8.3,0,0,0-2.6,5.9V208a8,8,0,0,0,8,8H208a8,8,0,0,0,8-8V115.5A8.3,8.3,0,0,0,213.4,109.6Z"
                      fill="none"
                      stroke="currentColor"
                      stroke-linecap="round"
                      stroke-linejoin="round"
                      stroke-width="16"
                    />
                  </svg>
                </div>
                <p
                  class="css-b717qx-Paragraph"
                >
                  Dashboard
                </p>
              </div>
            </div>
            <div
              class="css-14p1f3h-Flex"
            >
              <div
                class="css-4s1fvz-Flex"
              >
                <div
                  class="css-kvl0hh-Box"
                >
                  <svg
                    fill="currentColor"
                    height="24"
                    viewBox="0 0 256 256"
                    width="24"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <rect
                      fill="none"
                      height="256"
                      width="256"
                    />
                    <path
                      d="M168,224H56a8,8,0,0,1-8-8V72a8,8,0,0,1,8-8h80l40,40V216A8,8,0,0,1,168,224Z"
                      fill="none"
                      stroke="currentColor"
                      stroke-linecap="round"
                      stroke-linejoin="round"
                      stroke-width="16"
                    />
                    <path
                      d="M80,64V40a8,8,0,0,1,8-8h80l40,40V184a8,8,0,0,1-8,8H176"
                      fill="none"
                      stroke="currentColor"
                      stroke-linecap="round"
                      stroke-linejoin="round"
                      stroke-width="16"
                    />
                    <line
                      fill="none"
                      stroke="currentColor"
                      stroke-linecap="round"
                      stroke-linejoin="round"
                      stroke-width="16"
                      x1="88"
                      x2="136"
                      y1="152"
                      y2="152"
                    />
                    <line
                      fill="none"
                      stroke="currentColor"
                      stroke-linecap="round"
                      stroke-linejoin="round"
                      stroke-width="16"
                      x1="88"
                      x2="136"
                      y1="184"
                      y2="184"
                    />
                  </svg>
                </div>
                <p
                  class="css-b717qx-Paragraph"
                >
                  Documents
                </p>
              </div>
            </div>
            <div
              class="css-14p1f3h-Flex"
            >
              <div
                class="css-4s1fvz-Flex"
              >
                <div
                  class="css-kvl0hh-Box"
                >
                  <svg
                    fill="currentColor"
                    height="24"
                    viewBox="0 0 256 256"
                    width="24"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <rect
                      fill="none"
                      height="256"
                      width="256"
                    />
                    <path
                      d="M72,224H56a8,8,0,0,1-8-8V184"
                      fill="none"
                      stroke="currentColor"
                      stroke-linecap="round"
                      stroke-linejoin="round"
                      stroke-width="16"
                    />
                    <polyline
                      fill="none"
                      points="120 32 152 32 208 88 208 136"
                      stroke="currentColor"
                      stroke-linecap="round"
                      stroke-linejoin="round"
                      stroke-width="16"
                    />
                    <path
                      d="M48,64V40a8,8,0,0,1,8-8H80"
                      fill="none"
                      stroke="currentColor"
                      stroke-linecap="round"
                      stroke-linejoin="round"
                      stroke-width="16"
                    />
                    <polyline
                      fill="none"
                      points="152 32 152 88 208 88"
                      stroke="currentColor"
                      stroke-linecap="round"
                      stroke-linejoin="round"
                      stroke-width="16"
                    />
                    <path
                      d="M208,176v40a8,8,0,0,1-8,8h-8"
                      fill="none"
                      stroke="currentColor"
                      stroke-linecap="round"
                      stroke-linejoin="round"
                      stroke-width="16"
                    />
                    <line
                      fill="none"
                      stroke="currentColor"
                      stroke-linecap="round"
                      stroke-linejoin="round"
                      stroke-width="16"
                      x1="48"
                      x2="48"
                      y1="104"
                      y2="144"
                    />
                    <line
                      fill="none"
                      stroke="currentColor"
                      stroke-linecap="round"
                      stroke-linejoin="round"
                      stroke-width="16"
                      x1="112"
                      x2="152"
                      y1="224"
                      y2="224"
                    />
                  </svg>
                </div>
                <p
                  class="css-b717qx-Paragraph"
                >
                  Templates
                </p>
              </div>
            </div>
            <div
              class="css-14p1f3h-Flex"
            >
              <div
                class="css-4s1fvz-Flex"
              >
                <div
                  class="css-kvl0hh-Box"
                >
                  <svg
                    fill="currentColor"
                    height="24"
                    viewBox="0 0 256 256"
                    width="24"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <rect
                      fill="none"
                      height="256"
                      width="256"
                    />
                    <path
                      d="M122.7,25.9,42,42,25.9,122.7a8,8,0,0,0,2.2,7.2L132.5,234.3a7.9,7.9,0,0,0,11.3,0l90.5-90.5a7.9,7.9,0,0,0,0-11.3L129.9,28.1A8,8,0,0,0,122.7,25.9Z"
                      fill="none"
                      stroke="currentColor"
                      stroke-linecap="round"
                      stroke-linejoin="round"
                      stroke-width="16"
                    />
                    <circle
                      cx="84"
                      cy="84"
                      r="12"
                    />
                  </svg>
                </div>
                <p
                  class="css-b717qx-Paragraph"
                >
                  Tags
                </p>
              </div>
            </div>
            <div
              class="css-14p1f3h-Flex"
            >
              <div
                class="css-4s1fvz-Flex"
              >
                <div
                  class="css-kvl0hh-Box"
                >
                  <svg
                    fill="currentColor"
                    height="24"
                    viewBox="0 0 256 256"
                    width="24"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <rect
                      fill="none"
                      height="256"
                      width="256"
                    />
                    <polyline
                      fill="none"
                      points="152 32 152 88 208 88"
                      stroke="currentColor"
                      stroke-linecap="round"
                      stroke-linejoin="round"
                      stroke-width="16"
                    />
                    <polyline
                      fill="none"
                      points="148 128 172 152 148 176"
                      stroke="currentColor"
                      stroke-linecap="round"
                      stroke-linejoin="round"
                      stroke-width="16"
                    />
                    <polyline
                      fill="none"
                      points="108 128 84 152 108 176"
                      stroke="currentColor"
                      stroke-linecap="round"
                      stroke-linejoin="round"
                      stroke-width="16"
                    />
                    <path
                      d="M200,224a8,8,0,0,0,8-8V88L152,32H56a8,8,0,0,0-8,8V216a8,8,0,0,0,8,8Z"
                      fill="none"
                      stroke="currentColor"
                      stroke-linecap="round"
                      stroke-linejoin="round"
                      stroke-width="16"
                    />
                  </svg>
                </div>
                <p
                  class="css-b717qx-Paragraph"
                >
                  Reports
                </p>
              </div>
            </div>
            <div
              class="css-14p1f3h-Flex"
            >
              <div
                class="css-4s1fvz-Flex"
              >
                <div
                  class="css-kvl0hh-Box"
                >
                  <svg
                    fill="currentColor"
                    height="24"
                    viewBox="0 0 256 256"
                    width="24"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <rect
                      fill="none"
                      height="256"
                      width="256"
                    />
                    <path
                      d="M77.4,201.9l-32.3,27A8,8,0,0,1,32,222.8V64a8,8,0,0,1,8-8H216a8,8,0,0,1,8,8V192a8,8,0,0,1-8,8H82.5A7.8,7.8,0,0,0,77.4,201.9Z"
                      fill="none"
                      stroke="currentColor"
                      stroke-linecap="round"
                      stroke-linejoin="round"
                      stroke-width="16"
                    />
                  </svg>
                </div>
                <p
                  class="css-b717qx-Paragraph"
                >
                  Chat
                </p>
              </div>
            </div>
            <div
              class="css-14p1f3h-Flex"
            >
              <div
                class="css-4s1fvz-Flex"
              >
                <div
                  class="css-kvl0hh-Box"
                >
                  <svg
                    fill="currentColor"
                    height="24"
                    viewBox="0 0 256 256"
                    width="24"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <rect
                      fill="none"
                      height="256"
                      width="256"
                    />
                    <polyline
                      fill="none"
                      points="174 86 216 128 174 170"
                      stroke="currentColor"
                      stroke-linecap="round"
                      stroke-linejoin="round"
                      stroke-width="16"
                    />
                    <line
                      fill="none"
                      stroke="currentColor"
                      stroke-linecap="round"
                      stroke-linejoin="round"
                      stroke-width="16"
                      x1="104"
                      x2="216"
                      y1="128"
                      y2="128"
                    />
                    <path
                      d="M104,216H48a8,8,0,0,1-8-8V48a8,8,0,0,1,8-8h56"
                      fill="none"
                      stroke="currentColor"
                      stroke-linecap="round"
                      stroke-linejoin="round"
                      stroke-width="16"
                    />
                  </svg>
                </div>
                <p
                  class="css-b717qx-Paragraph"
                >
                  Logout
                </p>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="css-1bsurua-Flex"
      >
        <svg
          fill="currentColor"
          height="24"
          viewBox="0 0 256 256"
          width="24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <rect
            fill="none"
            height="256"
            width="256"
          />
          <line
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="16"
            x1="40"
            x2="216"
            y1="128"
            y2="128"
          />
          <line
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="16"
            x1="40"
            x2="216"
            y1="64"
            y2="64"
          />
          <line
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="16"
            x1="40"
            x2="216"
            y1="192"
            y2="192"
          />
        </svg>
      </div>
      <div
        class="css-1cadlib-Box"
      />
    </div>
  </div>
</div>
`;
